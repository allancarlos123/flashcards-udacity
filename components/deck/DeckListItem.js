import React, { Component } from 'react';
import { StyleSheet, Text } from 'react-native';

export default function DeckListItem( item ) {
  return (
      <Text style={styles.item}>{item.title}</Text>
  )
}

const styles = StyleSheet.create({
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});
