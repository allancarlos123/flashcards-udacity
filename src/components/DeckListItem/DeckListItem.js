import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text } from 'react-native';

const DeckListItem = ({ title, length}) => (
  <Text style={styles.item}>
    {title} ({length})
  </Text>
)

const styles = StyleSheet.create({
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});

DeckListItem.propTypes = {
  title: PropTypes.string.isRequired,
  length: PropTypes.number.isRequired
}

export default DeckListItem