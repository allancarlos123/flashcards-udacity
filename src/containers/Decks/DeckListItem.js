import React from 'react';
import { StyleSheet, Text } from 'react-native';

export default function DeckListItem( item ) {
  return (
      <Text style={styles.item}>{item.title} ({item.length})</Text>
  )
}

const styles = StyleSheet.create({
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});
