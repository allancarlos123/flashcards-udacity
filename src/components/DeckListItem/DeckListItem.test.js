import React from 'react'
import { create } from 'react-test-renderer'

import DeckListItem from './'

describe('DeckListItem', () => {
  const tree = create(
    <DeckListItem
      title="React"
      length={2}
    />
  );

  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
