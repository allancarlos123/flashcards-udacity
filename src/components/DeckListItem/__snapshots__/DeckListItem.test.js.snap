// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckListItem renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "fontSize": 18,
      "height": 44,
      "padding": 10,
    }
  }
>
  React
   (
  2
  )
</Text>
`;
